### ISSUE TITLE
Cop idea: avoid numbered variables unless needed

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#12392]
A cop idea for the `Naming` group: prefer `var` over `var_1` unless there's also `var_2` (etc.) in the same scope. Makes the code more readable:

```ruby
# bad
def foo
   user_1 = create(:user)
   user_1.destroy
end

# good
def foo
   user = create(:user)
   user.destroy
end

# good
def foo
   user_1 = create(:user)
   user_2 = create(:user)
   user_1.destroy
   user_2.destroy
end
```

**Comment by koic** [comment]
It's an interesting idea, but it cannot prevent false positives in cases with meaningful numbers. For example, in `Naming/VariableNumber`, certain method names are allowed by default, requiring a similar list of exceptions. However, managing such a list might be a hassle for users, leading to doubts about its value as a default cop.

**Comment by vlad-pisanov** [comment]
Maybe this could be a non-default option of `Naming/VariableNumber` so it can piggyback off the exception list, like:
```yaml
Naming/VariableNumber:
   Enabled: true
   OnlyWhenNeeded: true
```
